### PR TITLE
Fix suite freeze on non-existent xtrigger - 7.8.x

### PR DESCRIPTION
**Pull request by dwsutherland**
Port of [pull/3056](https://github.com/cylc/cylc/pull/3056)

### CODE REVIEW

**Comment by kinow** [comment]
Travis is unhappy because of an error with `pycodestyle` I believe.

```
lib/cylc/config.py:2107:33: E128 continuation line under-indented for visual indent
lib/cylc/config.py:2110:29: E128 continuation line under-indented for visual indent
The command "pycodestyle --ignore=E402,W503,W504 lib/cylc lib/Jinja2Filters/*.py lib/parsec/*.py $(grep -l '#!.*\<python\>' bin/*)" exited with 1.
```

(this error appears near the top of the output, which is misleading, as you normally expect the error near the bottom of the file/output)

**Comment by dwsutherland** [comment]
> (this error appears near the top of the output, which is misleading, as you normally expect the error near the bottom of the file/output)

I couldn't see that log detail at all, was just the working info.. anyways fixed.

**Comment by matthewrmshin** [comment]
Given the final comments under #3056, do we need to raise a new PR against master for a test and back port that here?

**Comment by matthewrmshin** [comment]
Happy to merge this otherwise, and have separate back port PR for test.

**Comment by matthewrmshin** [comment]
Is that #3060?

**Comment by dwsutherland** [comment]
> Given the final comments under #3056, do we need to raise a new PR against master for a test and back port that here?

Correct,  test PR to come post merge.

**Comment by kinow** [comment]
>Not here sorry; New test PR port of 3060 to come post merge of this one.

Yup, I merged it too quickly, sorry. So now we will have yet another backport pull request. Mea culpa :disappointed: I'll submit it once this gets merged.

**Comment by dwsutherland** [comment]
> > Not here sorry; New test PR port of 3060 to come post merge of this one.
> 
> Yup, I merged it too quickly, sorry. So now we will have yet another backport pull request. Mea culpa 😞 I'll submit it once this gets merged.

Worked out well, since your test construction > mine  ATM (>.<) ... Gave me a chance to use yours as a tutorial.

File: lib/cylc/config.py [hunk 1]
```diff
@@ -2100,6 +2101,15 @@ def load_graph(self):
                         if offset > old_offset:
                             self.taskdefs[task_name].xclock_label = label
                 else:
+                    try:
+                        if not callable(get_func(xtrig.func_name, self.fdir)):
+                            raise SuiteConfigError("ERROR, xtrigger function "
+                                                   "not callable: "
+                                                   "%s" % xtrig.func_name)
+                    except ImportError:
```
Comment (kinow) [inline lib/cylc/config.py]: Hmmm, just realized you have `ModuleNotFoundError` on `master`, and the parent class `ImportError` on `7.8.x`. I think we better use the parent `ImportError` on master too @dwsutherland . WDYT?

Comment (dwsutherland) [reply lib/cylc/config.py]: @kinow - Yes, as python3 gave me `ModuleNotFoundError`, but python2 produced `ImportError` and the former is not recognised. I didn't try `ImportError` in python3 (Is it back compat?), and figured both have diverged anyway.. Is this ok?

Comment (hjoliver) [reply lib/cylc/config.py]: Interesting (another hole in my Python 3 knowledge!).

Comment (kinow) [reply lib/cylc/config.py]: I believe `ImportError` is the parent class of `ModuleNotFoundError`.

>Interesting (another hole in my Python 3 knowledge!).

Me too! I thought at least this part was consistent in both. But catching the parent class should be fine, though better to confirm if that works. Perhaps the unit tests could be used to validate that it works?

Comment (dwsutherland) [reply lib/cylc/config.py]: > I believe `ImportError` is the parent class of `ModuleNotFoundError`.

Well I guess the child doesn't exist in python2  ... Consistency aside, would it have been better to use the parent?

Also does consistency matter between 7.8.x and 8? at some point we'll stop back porting changes? (as I guess it matters for that)

Comment (kinow) [reply lib/cylc/config.py]: Good point. If that works as-is, all good then I think

**Pull request merged**